src/stm32/Makefile

```diff
@@ -17,7 +17,12 @@ MCU := $(shell echo $(CONFIG_MCU))
 MCU_UPPER := $(shell echo $(CONFIG_MCU) | tr a-z A-Z | tr X x)
 
 CFLAGS-$(CONFIG_MACH_STM32F0) += -mcpu=cortex-m0 -Ilib/stm32f0/include
-CFLAGS-$(CONFIG_MACH_STM32F1) += -mcpu=cortex-m3 -Ilib/stm32f1/include
+ifeq ($(CONFIG_N32G452), y)
+	dirs-$(CONFIG_MACH_STM32F1) += lib/n32g45x
+	CFLAGS-$(CONFIG_MACH_STM32F1) += -mcpu=cortex-m4 -Ilib/stm32f1/include -Ilib/n32g45x/include
+else
+	CFLAGS-$(CONFIG_MACH_STM32F1) += -mcpu=cortex-m3 -Ilib/stm32f1/include
+endif
```

Comment:
I think this should be turned into lists.  Ifdefs in the makefile rapidly become unmaintainable.  Something like:
```
dirs-$(CONFIG_MACH_N32G452) += lib/n32g45x
```